src/libraries/System.Private.CoreLib/src/System/Collections/Generic/CollectionExtensions.cs

```diff
@@ -0,0 +1,164 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Collections.ObjectModel;
+using System.Diagnostics.CodeAnalysis;
+
+namespace System.Collections.Generic
+{
+    public static class CollectionExtensions
+    {
+        public static TValue? GetValueOrDefault<TKey, TValue>(this IReadOnlyDictionary<TKey, TValue> dictionary, TKey key) =>
+            dictionary.GetValueOrDefault(key, default!);
+
+        public static TValue GetValueOrDefault<TKey, TValue>(this IReadOnlyDictionary<TKey, TValue> dictionary, TKey key, TValue defaultValue)
+        {
+            if (dictionary is null)
```

Comment:
Why not use the one line helper anymore?

```C#
ArgumentNullException.ThrowIfNull(dictionary);
```

Reply:
Because the ThrowHelper pattern used in corelib is slightly more efficient, and the necessary method and ExceptionArgument value already exist, so this can just use them.